monasca_events_api/tests/test_stream_definitions.py

```diff
@@ -0,0 +1,402 @@
+
+    @mock.patch('monasca_events_api.v2.common.helpers.add_links_to_resource')
+    @mock.patch(
+        'monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch(
+        'monasca_events_api.common.repositories.mysql.streams_repository.StreamsRepository.create_stream_definition')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_expire_actions')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_fire_actions')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_description')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_name')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.StreamDefinitions._validate_stream_definition')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_json_msg_body')
+    @mock.patch(
+        'monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    def test_on_post_pass(
+            self,
+            validate,
+            tenantid,
+            json,
+            readjson,
+            streamvalid,
+            getname,
+            desc,
+            fireactions,
+            expire,
+            streamsrepo,
+            kafka,
+            addlink):
+        """POST method successful"""
+        validate.return_value = True
+        fireactions.return_value = "fire_actions"
+        getname.return_value = "Test"
+        addlink.return_value = "/v2.0/stream-definitions/{stream_id}"
+        expire.return_value = "expire_actions"
+        desc.return_value = "Stream_Description"
+        streamsrepo.return_value = True
+        readjson.return_value = {u'name': "Test",
+                                 u'id': "1",
+                                 u'description': "Stream_Description",
+                                 u'select': "select",
+                                 u'group_by': "group_by",
+                                 u'fire_criteria': "fire_criteria",
+                                 u'expiration': "expiration",
+                                 u'fire_actions': "fire_actions",
+                                 u'expire_actions': "expire_actions",
+                                 u'actions_enabled': u'true'}
+
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        streamvalid.return_value = True
+
+        streamsObj = StreamDefinitionsSubClass()
+        streamsObj._stream_definitions_repo = StreamsRepository()
+        streamsObj.stream_definition_event_message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        streamsObj.on_post(self._generate_req(), res)
+        self.assertEqual(falcon.HTTP_201, res.status)
```

Comment:
please don't mock streamvalid.  We want the validation code to execute, so I think remove:  streamvalid.return_value = True